topostats/io.py

```diff
@@ -149,16 +159,100 @@ def _load_jpk(self) -> None:
             LOGGER.error(f"[{self.filename}] : {exception}")
         return jpk
 
+    # uncomment below func and import libasd to use asd formats
+    #def load_asd(self) -> tuple:
```

Comment:
This shouldn't fail as the asd tests shouldn't be run.

This PR is basically here for the switch-up in how files are loaded and put into process scan. The features of loading asd formats are there but should be used with caution as the the libasd library isn't in setup.cfg because it the binaries aren't built on MacOS and so fails on some OS's.

As such I have commented out all asd references.

Reply:
First off, this is only going into dev because I thought that's where we are putting everything. Please feel free to push to PyPI when @SylviaWhittle's flattening is complete, this is not an essential requirement for the PyPI release.


So, to update you on this, I've taken your feedback and found a workaround for the running and tests. I've added try and excepts around the `libasd` import, functions and tests. These now make the code run as expected on non-libasd files, but throws the following warning if the module cannot be found (as an inpromptu solution), and does not halt the processing of the non-libasd files:
```
[Wed, 26 Oct 2022 14:46:44] [WARNING ] [topostats] libasd module not installed. Ignore if '.asd' files are not being processed. 
 Otherwise install via 'pip install .[libasd]' or if on an M1/2 Mac, follow libasd installation guidelines for 'build yourself' here: https://github.com/ToruNiina/libasd
```

libasd files will also throw this warning along with the ```NameError: name 'libasd' is not defined``` error when processed, thus stopping their processing until the module is imported.

As you can see, install instructions for the libasd files are found in the error message (and if you'd like to suggest somewhere else please let me know), with a pip installable `pip install .[libasd]` option for those without M1/M2 Macs, and the libasd instructions there if they do.

Tests have been handled in a similar way where `ModuleNotFoundError`'s are asserted true, and otherwise `AssertionError`'s are raised (if any). 

My ask of you, @ns-rse here, is if there is a way to also to `pip install .[libasd]` in the GitHub tests as these all use non-M1/2 MacOS's. This would solve the issue you present and run these tests in the GitHub workflows (and pass locally, although possibly not properly run depending on wether libasd is installed).

I'm also not sure of the possible complications this would induce to pushing to PyPI but I'm all ears :)